x/cdp/keeper/savings.go

```diff
@@ -40,10 +43,10 @@ func (k Keeper) DistributeSavingsRate(ctx sdk.Context, debtDenom string) error {
 		}
 		// (balance * rewardToDisribute) /  totalSupply
 		// interest is the ratable fraction of savings rate owed to that account, rounded using bankers rounding
-		interest := (sdk.NewDecFromInt(debtAmount).Mul(sdk.NewDecFromInt(surplusToDistribute))).Quo(sdk.NewDecFromInt(totalSupplyLessModAccounts.AmountOf(debtDenom))).RoundInt()
+		interest := (sdk.NewDecFromInt(debtAmount).Mul(totalSurplus)).Quo(totalSupply).RoundInt()
 		// sanity check, if we are going to over-distribute due to rounding, distribute only the remaining savings rate that hasn't been distributed.
-		if interest.GT(surplusToDistribute.Sub(surplusDistributed)) {
-			interest = surplusToDistribute.Sub(surplusDistributed)
+		if interest.GT(surplusToDistribute) {
```

Comment:
```suggestion
		interest = sdk.MinInt(interest, surplusToDistribute)
```

Reply:
Thanks, didn't know there was a `min` function for `sdk.Int` 👍